aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1335,56 +1417,41 @@ static inline Tensor& bmm_out_or_baddbmm_(Tensor& self_or_result, const Tensor&
       }
     }
   }
-  return self_or_result;
-}
-
-
-Tensor baddbmm_cpu(const Tensor& self, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha) {
-  Tensor result = at::empty({0}, self.options());
-  return at::native::baddbmm_out_cpu(self, batch1, batch2, beta, alpha, result);
+  return;
 }
 
-Tensor& baddbmm_out_cpu(const Tensor& self_, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha, Tensor &result) {
-  auto self = expand_size(self_, {batch1.size(0), batch1.size(1), batch2.size(2)}, "baddbmm");
-  result.resize_(self->sizes());
-  result.copy_(*self);
-  return at::native::baddbmm__cpu(result, batch1, batch2, beta, alpha);
-}
-
-Tensor& conjugate_mutable_input_if_needed(Tensor& self, bool conjugate) {
+void conjugate_mutable_input_if_needed(Tensor& self, bool conjugate) {
   if (conjugate) {
     self.conj_physical_();
   }
-  return self;
-}
-
-Tensor& baddbmm__cpu(Tensor& self, const Tensor& batch1, const Tensor& batch2, const Scalar& beta, const Scalar& alpha) {
-  bool self_is_conj = self.is_conj();
-  conjugate_mutable_input_if_needed(self, self_is_conj);
-  bmm_out_or_baddbmm_(self, batch1.resolve_conj(), batch2.resolve_conj(), beta, alpha, false);
-  return conjugate_mutable_input_if_needed(self, self_is_conj);
 }
 
-Tensor bmm_cpu(const Tensor& self, const Tensor& mat2) {
-  Tensor result = at::empty({0}, self.options());
-  return at::native::bmm_out_cpu(self, mat2, result);
-}
+TORCH_IMPL_FUNC(baddbmm_out_cpu)
+(const Tensor & self, const Tensor & batch1, const Tensor & batch2, const Scalar& beta, const Scalar& alpha, const Tensor& result) {
+    const Tensor& self_or_result = result;
+    if (result.is_same(self)) {
+      self_or_result.copy_(self);
+    }
+    bool self_is_conj = result.is_conj();
+    conjugate_mutable_input_if_needed(const_cast<Tensor&>(result), self_is_conj);
+    bmm_out_or_baddbmm_(const_cast<Tensor&>(result), batch1.resolve_conj(), batch2.resolve_conj(), beta, alpha, false);
+    conjugate_mutable_input_if_needed(const_cast<Tensor&>(result), self_is_conj);
```

Comment:
Can we propagate `const` here, instead of casting?

Reply:
Nice, done!